public/app/features/search/page/SearchPage.tsx

```diff
@@ -88,6 +88,14 @@ export default function SearchPage() {
     setSearchSelection(updateSearchSelection(searchSelection, !current, kind, [uid]));
   };
 
+  const onUpdateList = async () => {
+    // clean up search selection
+    setSearchSelection(clearSearchSelection(searchSelection));
```

Comment:
```suggestion
    setSearchSelection(newSearchSelection());
```

Maybe we could just reuse newSearchSelection here?